src/framework/components/animation/component.js

```diff
@@ -13,7 +13,9 @@ Object.assign(pc, function () {
      * @property {boolean} activate If true the first animation asset will begin playing when the scene is loaded.
      * @property {pc.Asset[]|number[]} assets The array of animation assets - can also be an array of asset ids.
      * @property {number} currentTime Get or Set the current time position (in seconds) of the animation.
-     * @property {number} duration Get the duration in seconds of the current animation.
+     * @property {number} duration Get the duration in seconds of the current animation. [read only]
+     * @property {pc.Skeleton} skeleton Get or Set the skeleton for the current model.
+     * @property {object<string, pc.Animation>} animations Get or Set dictionary of animations by name.

```

Comment:
The animation component was recently extended to support GLB animations. The result is unfortunately a bit messy: the component is either internally in the legacy 'json' animation mode or the newer 'glb' mode. In the legacy mode skeleton is valid, but in the newer glb mode skeleton is unused.

So it's probably not safe to document skeleton just yet (in any case it's a read-only property).

animations property is also read only.

Reply:
@slimbuckThank you for the feedback.

What's the equivalent to `skeleton` in "glb" mode? We're writing a glTF exporter and need to be able to properly access it, in order to export it.

I'm happy to change the documentation to read-only. But please note that both `skeleton` and `animations` have setters - according to [the schema](https://github.com/playcanvas/engine/blob/master/src/framework/components/animation/system.js#L2) which [`_buildAccessors` reads](https://github.com/playcanvas/engine/blob/master/src/framework/components/animation/system.js#L47) to create standard [getters and setters](https://github.com/playcanvas/engine/blob/master/src/framework/components/component.js#L34). Additionally, `AnimationComponent` even [listens for and handles changes to `animations` explicitly](https://github.com/playcanvas/engine/blob/master/src/framework/components/animation/component.js#L24).

Do you want me to still change both to read-only? I'll also add a comment about `skeleton` not being available when the asset is from glb/gltf.

Reply:
Hi @aidinabedi yup you're right, I forgot about the set_animations handler. It is clearly not read-only!

I guess we can keep skeleton with a comment as you suggest.

GLB mode uses the AnimController and related classes (see anim.js). These are all currently private and some of the API is likely to change in future. So we can't document them just yet.